### ISSUE TITLE
Incorrect message on UI when podGC is enabled with `deleteDelayDuration`

### ISSUE THREAD

**Issue by bkanuka** [argoproj/argo-workflows#12395]
# Summary

I have my `workflowDefaults` set with:
```yaml
      podGC:
        strategy: OnWorkflowCompletion
        deleteDelayDuration: 168h  # Delete pods 7 days after completion
```

However, the UI states:  "Your pod GC settings will delete pods and their logs immediately on completion. ".  This isn't quite true because my pods will be deleted 7 days after completion.

I would propose that this message is _not_ shown when the `podGC` is set to anything other than the default, or greater than the default.

## Use Cases

I use `OnWorkflowCompletion` and a long `deleteDelayDuration` so that my Kubernetes cluster isn't forever holding on to a growing list of completed pods.  Keeping only the last week of pods is sufficient and keeps the list of pods in k8s at a reasonable level.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by agilgur5** [comment]
Ah yea, Workflow-level `deleteDelayDuration` is a newer feature I added in #11325 that released in 3.5, so that message pre-dates the existence of the feature.

Would you like to contribute a fix for the message when `deleteDelayDuration` is used?

**Comment by bkanuka** [comment]
I'm probably not the best person. Sorry...but I do suspect it's a great
first issue!
